plugins/parsers/json/parser_test.go

```diff
@@ -448,22 +449,148 @@ func TestJSONParseNestedArray(t *testing.T) {
 	"total_devices": 5,
 	"total_threads": 10,
 	"shares": {
-	"total": 5,
-	"accepted": 5,
-	"rejected": 0,
-	"avg_find_time": 4,
-	"tester": "work",
-	"tester2": "don't want this",
-	"tester3": 7.93
+		"total": 5,
+		"accepted": 5,
+		"rejected": 0,
+		"avg_find_time": 4,
+		"tester": "work",
+		"tester2": "don't want this",
+		"tester3": {
+			"hello":"sup",
+			"fun":"money",
+			"break":9
+		}
 	}
 	}`
 
 	parser := JSONParser{
 		MetricName: "json_test",
-		TagKeys:    []string{"total_devices", "total_threads", "shares_tester", "shares_tester3"},
+		TagKeys:    []string{"total_devices", "total_threads", "shares_tester3_fun"},
 	}
 
 	metrics, err := parser.Parse([]byte(testString))
+	log.Printf("m[0] name: %v, tags: %v, fields: %v", metrics[0].Name(), metrics[0].Tags(), metrics[0].Fields())
 	require.NoError(t, err)
 	require.Equal(t, len(parser.TagKeys), len(metrics[0].Tags()))
 }
+
+func TestJSONQueryErrorOnArray(t *testing.T) {
+	testString := `{
+		"total_devices": 5,
+		"total_threads": 10,
+		"shares": {
+			"total": 5,
+			"accepted": 6,
+			"test_string": "don't want this",
+			"test_obj": {
+				"hello":"sup",
+				"fun":"money",
+				"break":9
+			},
+			"myArr":[4,5,6]
+		}
+	}`
+
+	parser := JSONParser{
+		MetricName: "json_test",
+		TagKeys:    []string{},
+		JSONQuery:  "shares.myArr",
+	}
+
+	_, err := parser.Parse([]byte(testString))
+	assert.Error(t, err)
```

Comment:
Use require instead of assert (almost) everywhere.  Doesn't really matter here but this will prevent confusing chained errors.